src/allotropy/parsers/beckman_pharmspec/beckman_pharmspec_parser.py

```diff
@@ -0,0 +1,262 @@
+            items.append(DistributionItem(**item))
+        # TODO get test example for data_processing_omission_setting
+        dd = DistributionDocumentItem(distribution=items)
+        return [dd]
+
+    def _create_calculated_document_items(
+        self, df: pd.DataFrame, feature: str
+    ) -> list[CalculatedDataDocumentItem]:
+        cols = column_map.values()
+        items = []
+        for row in df.index:
+            for col in [x for x in cols if x in df.columns]:
+                prop = col.replace(
+                    " ", "_"
+                )  # to be able to set the props on the DistributionItem
+                items.append(
+                    CalculatedDataDocumentItem(
+                        calculated_data_name=f"{feature}_{prop}".lower(),
+                        calculated_result=get_property_from_sample(
+                            prop, df.at[row, col]
+                        ),
+                        data_source_aggregate_document=TDataSourceAggregateDocument(
+                            data_source_document=[
+                                DataSourceDocumentItem(
+                                    data_source_identifier=random_uuid_str(),
+                                    data_source_feature=col,
+                                )
+                            ]
+                        ),
+                    )
+                )
+        return items
+
+    def _get_software_version_report_string(self, report_string: str) -> str:
+        match = re.search(r"v(\d+(?:\.\d+)?(?:\.\d+)?)", report_string)
+        if match:
+            return match.group(1)
+        return "Unknown"
+
+    def _setup_model(self, df: pd.DataFrame, file_name: str) -> Model:
+        """Build the Model
+
+        :param df: the raw dataframe
+        :return: the model
+        """
+        data = self._extract_data(df)
+        measurement_doc_items = []
+        calc_agg_doc = None
+        for g, gdf in data.groupby("Run No."):
+            name = str(g)
+            if g in VALID_CALCS:
+                calc_agg_doc = TCalculatedDataAggregateDocument(
+                    calculated_data_document=self._create_calculated_document_items(
+                        gdf, name
+                    )
+                )
+            else:
+                measurement_doc_items.append(
+                    MeasurementDocumentItem(
+                        measurement_identifier=name,
+                        measurement_time=pd.to_datetime(
+                            str(df.at[8, 5]).replace(".", "-")
+                        ).isoformat(timespec="microseconds")
+                        + "Z",
+                        device_control_aggregate_document=DeviceControlAggregateDocument(
+                            device_control_document=[
+                                DeviceControlDocumentItemModel(
+                                    flush_volume_setting=TQuantityValueMilliliter(0),
```

Comment:
Just confirming: we are purposely setting this to 0 because it is required by the schema, but not provided?

Reply:
That's correct.